packages/node/test/manual/release-health/application-mode-session.js

```diff
@@ -0,0 +1,72 @@
+const Sentry = require('../../../dist');
+
+function assertSessions(actual, expected) {
+  if (JSON.stringify(actual) !== JSON.stringify(expected)) {
+    console.error('FAILED: Sessions do not match');
+    process.exit(1);
+  }
+}
+
+function constructStrippedSessionObject(actual) {
+  const { init, status, errors, release, did } = actual;
+  return { init, status, errors, release, did };
+}
+
+let remaining = 2;
+
+class DummyTransport {
+  sendSession(session) {
+    if (remaining === 2) {
+      assertSessions(constructStrippedSessionObject(session),
+        {
+          init: true,
+          status: "ok",
+          errors: 1,
+          release: "2.0.x",
+          did: "ahmed2",
+        }
+      )
+    }
+    else {
+      assertSessions(constructStrippedSessionObject(session),
+        {
+          init: true,
+          status: "exited",
+          errors: 1,
+          release: "2.0.x",
+          did: "ahmed2",
+        }
+      )
+    }
+
+    --remaining;
+
+    if (!remaining) {
+      console.log('SUCCESS: All application mode sessions were sent to node transport as expected');
+      process.exit(0);
+    }
+  }
+  close() {
+    return Promise.resolve({
+      status: 'success',
+    });
+  }
+}
+
+Sentry.init({
+  dsn: 'http://test@example.com/1337',
+  release: '1.1',
+  transport: DummyTransport,
+});
+
+const hub = Sentry.getCurrentHub();
+
+// Start the session
+hub.startSession({ user: { username: 'ahmed' } });
+// endSession should be called at the exit of a process (very end of all logic)
+process.on('exit', () => hub.endSession());
+
+// Update Session Object
+hub.getScope().getSession().update({ release: '2.0.x', did: 'ahmed2' });
```

Comment:
mutating `did` and `release` violates the spec, https://develop.sentry.dev/sdk/sessions/#attribute-immutability

Reply:
Resolved!